### PR TITLE
feat: skip chart prep for local

### PR DESCRIPTION
**Pull request by zhaque44**
resolved: https://github.com/helmfile/helmfile/pull/1664

### CODE REVIEW

File: pkg/app/run.go [hunk 1]
```diff
@@ -39,6 +39,19 @@ func (r *Run) askForConfirmation(msg string) bool {
 	return AskForConfirmation(msg)
 }
 
+func (r *Run) prepareChartsIfNeeded(helmfileCommand string, dir string, concurrency int, opts state.ChartPrepareOptions) (map[state.PrepareChartKey]string, error) {
```
Comment (yxxhero) [inline pkg/app/run.go:42]: concurrency? it seem like not be handled?

File: pkg/app/run.go [hunk 2]
```diff
@@ -39,6 +39,19 @@ func (r *Run) askForConfirmation(msg string) bool {
...
+	if strings.EqualFold(helmfileCommand, "write-values") || strings.EqualFold(helmfileCommand, "list") {
```
Comment (yxxhero) [inline pkg/app/run.go:43]: we can use slices.Contains func.

File: pkg/app/run.go (hunk 1 above)
Comment (zhaque44) [reply pkg/app/run.go:43]: Now, wherever this function is called, valid concurrency values should be (greater than 0) are always provided. this is handled now

File: pkg/app/run.go [hunk 3]
```diff
@@ -39,6 +41,26 @@ func (r *Run) askForConfirmation(msg string) bool {
 	return AskForConfirmation(msg)
 }
 
+func (r *Run) prepareChartsIfNeeded(helmfileCommand string, dir string, concurrency int, opts state.ChartPrepareOptions) (map[state.PrepareChartKey]string, error) {
+	if concurrency <= 0 {
+		log.Printf("Warning: invalid concurrency value: %d, defaulting to 1", concurrency)
+		concurrency = 1
+	}
```
Comment (yxxhero) [inline pkg/app/run.go:48]: @zhaque44 oh. sorry. we should remvoe the code on the above.

**Review by yxxhero — APPROVED** [review]

**Comment by uristernik** [comment]
Sorry, I have exams in Uni, didn't get to it 🙏

**Comment by uristernik** [comment]
Thanks for taking it

**Comment by zhaque44** [comment]
> Thanks for taking it

of course! cheers... @yxxhero @uristernik

**Pull request merged**